### ISSUE TITLE
UPDATE EDGE statement [moved]

### ISSUE THREAD

**Issue by lvca** [orientechnologies/orientdb#1114]
This is [Issue 1114](http://code.google.com/p/orient/issues/detail?id=1114) moved from a Google Code project.
Added by 2012-10-14T22:09:41.000Z by [infected...@gmail.com](http://code.google.com/u/105221022043430137887/).
Please review that bug for more context and additional comments, but update this bug.

Original labels: Type-Enhancement, Priority-Low, v2.1.0

### Original description

    Having an UPDATE EDGE statement would be nice because there is less overhead then deleting and recreating the edge.
    
    Discussion here: https://groups.google.com/forum/#!topic/orient-database/Z5ZElaxVSZ0

**Comment by lucasxteixeira** [comment]
+1

**Comment by thedanielhanke** [comment]
+1

**Comment by luigidellaquila** [comment]
Same for UPDATE VERTEX

**Comment by AlexFiverr** [comment]
+1

**Comment by yujinlim** [comment]
+1

**Comment by iforwms** [comment]
+1

**Comment by AlexFiverr** [comment]
hey, any updates on the version this feature will be in? 
thanks!

**Comment by pisaruk** [comment]
+1

**Comment by StarpTech** [comment]
+1

**Comment by liorn** [comment]
+1

**Comment by FGRibreau** [comment]
:+1:

**Comment by sebastiandev** [comment]
+1

**Comment by smolinari** [comment]
:+1:

**Comment by matanshukry** [comment]
As far as I can tell, update works (including in/out):
```
UPDATE FRIEND SET in=#11:5 WHERE in.name="Samantha" and out.name="Mat"
```

Although, using a query inside the set clause for in/out will cause it to return array, which is not the desired behavior. Example:
```
UPDATE Friend SET in=(SELECT FROM User WHERE name="Jason") WHERE in.name="Samantha" and out.name="Mat"
```
When later querying the table Friend, we can see column "in" will contain [#11:4], emphasis on the brackets [], which means an array.


Upsert also works, although when creating a new vertex it doesn't set the in/out properties.
You could set the in/out properties your self, like this:
```
UPDATE Friend SET comment="Wazzzaup", in=#11:5, out=#11:6 UPSERT WHERE in.name="Jason" AND out.name="Mat"
```
It will result in a longer query when using sub-query for ```in=``` and ```out=```, but at least it works (sub-query has same problem as above).

To summarize, the only problem I can see is using a sub-query for in/out. So:

1. Is there any other feature required here? If not, can we agree the UPDATE command is suffice and there is no need for an UPDATE EDGE command?
2. Is there anyway to set the in/out properties, perhaps one that I missed?

**Comment by smolinari** [comment]
> 2. Is there anyway to set the in/out properties, perhaps one that I missed?

That is the reason for the suggestion, I believe. That, and also a way to change the edge type with one command. Whereas, I see that also as a rare "edge" use case. LOL! 

Scott

**Comment by matanshukry** [comment]
Added a PR that resolves the sub-query issues.

Assuming it is accepted, the only reason to keep this issue would be to create a command that could change the edge type.

In my opinion, even if there should be a command to change it, UPDATE EDGE should not be the name of that command. It's a bit misleading, since we are really changing the edge it self, and not simply updating it's properties.

**Comment by smolinari** [comment]
Maybe `MODIFY EDGE`?

Scott

**Comment by lvca** [comment]
An `UPDATE EDGE` command is supposed to maintain the integrity of the graph. if you just update in/out, vertices could refer to wrong edges. So the command should do much more.

**Comment by smolinari** [comment]
So, then :+1: :+1: for me! :smile: 

Scott

**Comment by luigidellaquila** [comment]
Hi guys,

happy to announce that I implemented this in develop branch (2.2).
Current syntax is exactly the same as normal update, so eg. (reverting an edge)

```
UPDATE EDGE EdgeClass SET
out = #12:0
in = #12:1
WHERE out = #12:1
and in = #12:0
```

The implementation validates that 
- both new out and in are vertices
- the target is an edge

We are not in RC yet, so if you think UPDATE EDGE is not the most appropriate name, please vote for another one ;-)

I'm waiting for final confirmation to update the docs

Thanks

Luigi

**Comment by lvca** [comment]
sounds good to me, +1

**Comment by michael-go** [comment]
@luigidellaquila does it also support UPSERT? 

Thanks!

**Comment by luigidellaquila** [comment]
Hi @michael-go 

no, right now it doesn't, but I'll work on that in the near future.
Could you please open a separate issue for that?

Thanks

Luigi

**Comment by michael-go** [comment]
I see now that a ticket for Edge UPSERT already exists: https://github.com/orientechnologies/orientdb/issues/4436

Thanks

**Comment by luigidellaquila** [comment]
Yes, you are right, thank you for the pointer!

Luigi
